flow/templates/base_script.sh

```diff
@@ -16,15 +16,12 @@ cd {{ project.config.project_dir }}
 {% block body %}
 {% set cmd_suffix = cmd_suffix|default('') ~ (' &' if parallel else '') %}
 {% for operation in operations %}
-{% if operation.directives.nranks and not mpi_prefix %}
-{% set mpi_prefix = "%s -n %d "|format(mpiexec|default("mpiexec"), operation.directives.nranks) %}
-{% endif %}
 
 # {{ "%s"|format(operation) }}
 {% if operation.directives.omp_num_threads %}
 export OMP_NUM_THREADS={{ operation.directives.omp_num_threads }}
 {% endif %}
-{{ mpi_prefix }}{{ cmd_prefix }}{{ operation.cmd }}{{ cmd_suffix }}
+{{ mpi_prefix|default(operation|get_prefix) }}{{ cmd_prefix }}{{ operation.cmd }}{{ cmd_suffix }}
```

Comment:
Since the environment provides a generic `cmd_prefix`, this should probably operate on `cmd_prefix`, correct? And `mpi_prefix` is obsolete?

Reply:
In current version flow allows the user to provide `mpi_prefix` and `cmd_prefix` separately which allows the user to add additional `cmd_prefix` but still use flow to get proper mpi_prefix.
For now, for the purpose of backward compatibility we think it best to keep both now and the combined prefix is simply called `prefix`. But for future when breaking change is allowed we can maybe change to another consensus that `cmd_prefix` should include all if user were to provide one.